docs/ocis/development/extensions.md

```diff
@@ -7,227 +7,70 @@ geekdocEditPath: edit/master/docs/ocis/development
-        "myconfig": "is awesome"
-      }
-    }
-  ]
-}
-```
+One main point you should get about storage in an oCIS extension is that you should never use the filesystem, but always use the CS3 APIs.
 
-## ownCloud Web extension points
+## Development Roadmap
 
-{{< hint info >}}
-For an up to date list check out [the ownCloud Web documentation](https://github.com/owncloud/web/issues/2423).
-{{< /hint >}}
+When reading the above section you might have noticed, that some integrations need some manual setup as of now but we are planning to switch to automatic service discovery in a later development stage of oCIS.
 
-Several ones available:
+There are some more topics on our roadmap for the extension system:
 
-### ownCloud Web
-- App switcher (defined in config.json)
-- App container (loads UI of your extension)
-
-### Files app
-- File action
-- Create new file action
-- Sidebar
-- Quick access for sidebar inside of file actions (in the file row)
-
-Example of a file action in the `app.js`:
-```js
-const appInfo = {
-  name: 'MarkdownEditor',
-  id: 'markdown-editor',
-  icon: 'text',
-  isFileEditor: true,
-  extensions: [{
-    extension: 'txt',
-    newFileMenu: {
-      menuTitle ($gettext) {
-        return $gettext('Create new plain text file…')
-      }
-    }
-  },
-  {
-    extension: 'md',
-    newFileMenu: {
-      menuTitle ($gettext) {
-        return $gettext('Create new mark-down file…')
-      }
-    }
-  }]
-}
-```
-
-For the side bar have a look at the files app, `defaults.js` & `fileSideBars`
-
-## API driven development
-
-Until now we only had a look at the ui and how the extensions are managed on the cli. But how do apps actually talk to the server?
-
-Short answer: any way you like
-
-Long answer: micro and ocis-hello follow a protocol driven development:
-
-- specify the API using protobuf
-- generate client and server code
-- evolve based on the protocol
-
-- CS3 api uses protobuf as well and uses GRPC
-
-- ocis uses go-micro, which provides http and grpc gateways
-- the gateways and protocols are optional
-
-- owncloud and kopano are looking into a [MS graph](https://developer.microsoft.com/de-de/graph) like api to handle ownCloud Web requests.
-  - they might be about user, contacts, calendars ... which is covered by the graph api
-  - we want to integrate with eg. kopano and provide a common api (file sync and share is covered as well)
-
-- as an example for protobuf take a look at [ocis-hello](https://github.com/owncloud/ocis-hello/tree/master/pkg/proto/v0)
+- Events: allow extensions to register for some events. On occurrence of the event they will receive some kind of notification or a callback. This would for example allow your application to perform actions on the creations, modification or deletion of a file.
+- Entitlement of extensions to act in behalf of users without a request context: user should be able to grant and revoke extensions access to their files even in behalf of them eg. in the background.
```

Comment:
This point confuses me a bit and seems to mix two things into one

```suggestion
- Impersonation: Users should be able to grant and revoke extensions the ability of acting on their behalf, e.g. for scheduled operations in the background.
```

Reply:
Currently an extension can take the user authentication from a request to do something with the users files.
If an extension wants to interact with files of a user without that request context, it currently needs to know the REVA JWT secret, but then the extension is able access ALL users files.

Therefore we need a concept which allows an extension to access files of one user without granting access to all other users. Basically something like when you add an app to your Github account which is allowed to see your private repositories

Reply:
Hmmm totally get it, not sure how to communicate it in a simple & terse way to somebody not that deep into technical details. Since this section is about what's coming in the future I'd focus on the second part of your reply here and give a concrete example of what it could be used for :)

Reply:
I decided to remove the development roadmap completely. Imho a section like this should be heavily backed by tickets but I couldn't find any usefull.